modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/IgniteTableImpl.java

```diff
@@ -369,6 +370,16 @@ public <RowT> CompletableFuture<?> upsertAll(
         return CompletableFuture.allOf(futures);
     }
 
+    private static List<ByteBuffer> serializeBinaryRows(Collection<BinaryRow> rows) {
+        var result = new ArrayList<ByteBuffer>(rows.size());
+
+        for (BinaryRow row : rows) {
+            result.add(row.byteBuffer());
```

Comment:
In the future,  I guess we would like to remove this "byteBuffer" method. In that case it would be more convenient to have specific BinaryRowMessage class, or something similar, that would have schema version and the payload in different fields.
Should we do it right now, or let's wait until the method is removed?

Reply:
Because what's the point of making a copy of an array if all we want is to slap a single integer to it. That's extra copying that most likely affects the performance in negative way.